src/bootchooser.c

```diff
@@ -1456,6 +1458,326 @@ static gboolean efi_get_state(RaucSlot* slot, gboolean *good, GError **error)
 	return TRUE;
 }
 
+// TODO add efibootguard bootchooser tests
+
+// Getting slots:
+// TODO call "bg_printenv --raw --part=1" & return k=>v array
+// TODO call "bg_printenv --raw --current"
+// TODO --part=1 can be "bootname" from config file
+
+// Setting slots:
+// TODO
+
+// #define BG_PRINTENV_NAME "bg_printenv"
+// #define BG_SETENV_NAME "bg_setenv"
+
+static gboolean efibootguard_env_get(const gchar *slot, const gchar *key, GString **value, GError **error)
+{
+	g_autoptr(GPtrArray) sub_args = NULL;
+	g_autoptr(GSubprocess) sub = NULL;
+	GError *ierror = NULL;
+	g_autoptr(GBytes) sub_stdout_buf = NULL;
+	const char *sub_stdout;
+	gsize sub_stdout_size;
+	gsize offset;
+	gsize size;
+	gint ret;
+
+	g_return_val_if_fail(slot, FALSE);
+	g_return_val_if_fail(key, FALSE);
+	g_return_val_if_fail(value && *value == NULL, FALSE);
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+	sub_args = g_ptr_array_new_full(5, g_free);
+	g_ptr_array_add(sub_args, g_strdup(BG_PRINTENV_NAME));
+	g_ptr_array_add(sub_args, g_strdup("--raw"));
+	g_ptr_array_add(sub_args, g_strdup("-p"));
+	g_ptr_array_add(sub_args, g_strdup(slot));
+	g_ptr_array_add(sub_args, NULL);
+
+	sub = r_subprocess_newv(sub_args, G_SUBPROCESS_FLAGS_STDOUT_PIPE | G_SUBPROCESS_FLAGS_STDERR_MERGE, &ierror);
```

Comment:
makes sense to rework this function to get of the data all at once perhaps returning a struct